### ISSUE TITLE
RPLIDAR's filtering should be rethought

### ISSUE THREAD

**Issue by GNSS-Stylist** [GNSS-Stylist/GNSS-Stylus#7]
RPLIDAR (A3M1) returns erroneous measurements when there's rapid change in measured distances. This is especially true when the nearer object's width is narrow (probably threshold of width being the distance between the laser and the "triangulating" receiver when the unit can "see though" (laser shooting and receiver seeing the "laser dot" on other sides of the nearer object) the nearer object). This causes artifacts into the point cloud.

It should also be checked and taken into account if RPLIDAR returns out-of-order measurements (= not monotonically increasing angles). If measurements are not monotonically increasing, this might also be used as a some kind of clue in filtering. This may also be related to "Distance delta limit" and "Distance slope limit"-parameters not working as expected.

Might also be worth considering using some other type of unit, maybe preferably giving more raw measurements so that the filtering could be done in "deeper lever".

TL;DR: Check what the lidar does and make the filtering better.

**Comment by GNSS-Stylist** [comment]
Read https://github.com/GNSS-Stylist/GNSS-Stylus/commit/9f233c82a7d19478b00614a69ba29c1cc6414f9c for some more thoughts about this.
